controllers/cluster_controller_phases_test.go

```diff
@@ -50,7 +50,7 @@ func TestClusterReconcilePhases(t *testing.T) {
 					Port: 8443,
 				},
 				InfrastructureRef: &corev1.ObjectReference{
-					APIVersion: "infrastructure.cluster.x-k8s.io/v1alpha2",
```

Comment:
think it's worth replacing these with `infrav1.GroupVersion.String()`?

Reply:
Probably not here, since the api version here should be pretty arbitrary, and would otherwise require pulling in a dependency on an actual infrastructure provider to make the call.

Reply:
mm, yeah, that makes sense